Tests/Pcap++Test/Tests/LiveDeviceTests.cpp

```diff
@@ -379,91 +380,209 @@ PTF_TEST_CASE(TestPcapLiveDeviceStatsMode)
-	liveDev->stopCapture();
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, nullptr, 1), -1);
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeNoTimeout, &packetCount, 30), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+	// open device
+	pcpp::PcapLiveDevice::DeviceConfiguration newConfig;
+	newConfig.usePoll = true;
 
-	// verify it's possible to capture non-blocking mode after blocking mode
-	packetCount = 0;
-	PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
+	PTF_ASSERT_TRUE(liveDev->open(newConfig));
+	DeviceTeardown devTeardown(liveDev);
 
-	int totalSleepTime = 0;
-	while (totalSleepTime <= 5)
+	int packetCount = 0;
+
+	auto func = [&]() -> int
 	{
-		pcpp::multiPlatformSleep(1);
-		totalSleepTime += 1;
-		if (packetCount > 0)
-			break;
-	}
+		return liveDev->startCaptureBlockingMode(packetArrivesBlockingModeTimeout, &packetCount, 2); // ideally, it should timeout after 2 seconds
+	};
 
-	liveDev->stopCapture();
+	// test if it will get timeout after 2 seconds
+	std::packaged_task<int()> task(func);
 
-	PTF_PRINT_VERBOSE("Total sleep time: " << totalSleepTime << " secs");
+	std::future<int> future = task.get_future();
 
-	PTF_ASSERT_GREATER_THAN(packetCount, 0);
+	// Start a new thread to execute `startCaptureBlockingMode` and to test it timeout
+	std::thread thread(std::move(task));
 
-	// verify it's possible to capture blocking mode after non-blocking mode
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeNoTimeout, &packetCount, 30), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+	// Wait for the function to finish or the timeout to occur
+	auto status = future.wait_for(std::chrono::milliseconds(2050)); // a little bit more than 2000ms, to check it is timeout
 
-	// try to start capture from within the callback, verify no error
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStartCapture, &packetCount, 30), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+	// the function got a timeout, due to it is still blocking
+	PTF_ASSERT_TRUE(status == std::future_status::ready);
 
-	// try to stop capture from within the callback, verify no impact on capturing
-	packetCount = 0;
-	PTF_ASSERT_EQUAL(liveDev->startCaptureBlockingMode(packetArrivesBlockingModeStopCapture, &packetCount, 10), 1);
-	PTF_ASSERT_EQUAL(packetCount, 5);
+	PTF_ASSERT_EQUAL(future.get(), -1); // function timeout correctly
+	PTF_ASSERT_EQUAL(packetCount, 0);
+	liveDev->close();
+#else
+	PTF_SKIP_TEST("This test can not run in Windows environment");
+#endif
+} // TestPcapLiveDeviceBlockingMode
 
-	// verify it's possible to capture non-blocking after the mess done in previous lines
-	packetCount = 0;
-	PTF_ASSERT_TRUE(liveDev->startCapture(packetArrives, &packetCount));
+PTF_TEST_CASE(TestPcapLiveDeviceBlockingModeNotTimeoutWithoutPoll)
+{
+#if !defined(_WIN32) and defined(MANUAL_TEST)
```

Comment:
same for this test. use `MANUAL_TEST` here.

Reply:
I think we can remove this test also...